src/pages/publication/public_publication/PublicPublication.tsx

```diff
@@ -7,24 +7,32 @@ import useFetchPublication from '../../../utils/hooks/useFetchPublication';
 import PublicPublicationContent from './PublicPublicationContent';
 import NotPublished from '../../errorpages/NotPublished';
 import NotFound from '../../errorpages/NotFound';
+import { useSelector } from 'react-redux';
+import { RootStore } from '../../../redux/reducers/rootReducer';
 
 const PublicPublication: FC = () => {
   const { identifier } = useParams();
   const [publication, isLoadingPublication] = useFetchPublication(identifier);
   const history = useHistory();
+  const user = useSelector((store: RootStore) => store.user);
 
   useEffect(() => {
-    history.replace(`/publication/${identifier}/public`, {
-      title: publication?.entityDescription?.mainTitle,
-    });
+    if (publication) {
+      history.replace(`/publication/${identifier}/public`, {
+        title: publication.entityDescription.mainTitle,
+      });
+    }
   }, [publication, history, identifier]);
 
+  const isAllowed =
```

Comment:
isAllowed to do what?

Reply:
`isAllowedToSeePublicPublication`? ble liksom så langt, men kan endre til det hvis du ikke har et bedre forslag

Reply:
ja, jeg synes det er bedre 👍